File: October-17-2025/containerMostWater.js

var height = [1,7,2,5,4,7,3,6]



// Brute Force Method: 

const area = (height) => {
  let res = 0;

  for(let left = 0; left < height.length; left++){
    for(let right = left + 1; right < height.length; right++){
      let area = (right-left) * Math.min(height[left], height[right])
      res = Math.max(res, area)
    }
  }
  return res
}

// console.log(area(height))

// Linear method

const areaII = (height) => { 
  let left = 0; 
  let right = height.length - 1
  let maxArea = 0; 

  while(left < right){
    let area = (right - left) * Math.min(height[left], height[right])
    maxArea = Math.max(maxArea, area)
    
    if(height[left] < height[right]){ 
      left++
    } else { 
      right--
    }
  }
  return maxArea
}


console.log(areaII(height))

File: 2025/November-7-2025/containerMostWater.py
from typing import List

height = [1,7,2,5,4,7,3,6]


def largestContainer(height: List[int]) -> int: 
  left = 0
  right = len(height) -1
  max_area = 0

  while left < right:
    width = right - left
    area = min(height[left], height[right]) * width

    max_area = max(max_area, area)

    if height[left] < height[right]:
      left+=1 
    else: 
      right-=1
  return max_area

print(largestContainer(height))